ChartDemo/Features/CandleStickChart/ViewController/CandleStickChartViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  CandleStickChartViewController.swift
+//  ChartDemo
+//
+//  Created by Fani on 20.10.22.
+//  Copyright © 2022 Randhir Kumar. All rights reserved.
+//
+
+import UIKit
+import Charts
+
+class CandleStickChartViewController: UIViewController {
+    @IBOutlet private weak var candleStickChartView: CandleStickChartView!
+    @IBOutlet private weak var animationSegmentControl: UISegmentedControl!
+    
+    private var hideChartData = false
+    var animationSelected: AnimationOptions = .linear {
+        didSet {
+            updateChartData()
+            setCandleChartData()
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = ChartType.candleStickChart.rawValue
+        setCandleChartData()
+    }
+
+    @IBAction func segmentAction(_ sender: UISegmentedControl) {
+        switch sender.selectedSegmentIndex {
+        case 0:
+            animationSelected = .linear
+        case 1:
+            animationSelected = .easeInCubic
+        default:
+            animationSelected = .easeOutCubic
+        }
+        hideChartData = true
+    }
+    
+    func updateChartData()  {
+        if hideChartData {
+            hideChartData = false
+            candleStickChartView.data = nil
+            return
+        }
+    }
+    
+    func setCandleChartData() {
+        candleStickChartView.autoScaleMinMaxEnabled = true
+        setCandleDataCount(8, range: 30)
+    }
+    
+    func setCandleDataCount(_ count: Int, range: UInt32) {
```

Comment:
Here too , can you please mark this private func

Reply:
Marked as private.